### PR TITLE
nixos/tmp: move /tmp options under boot.tmp

### PR DESCRIPTION
**Pull request by SuperSandro2000**
Open for suggestions for better names

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/system/boot/tmp.nix [hunk 1]
```diff
@@ -3,62 +3,60 @@
...
         Whether to delete all files in {file}`/tmp` during boot.
       '';
     };
 
-    boot.tmpOnTmpfs = mkOption {
+    boot.tmp.onTmpfs = mkOption {
       type = types.bool;
       default = false;
       description = lib.mdDoc ''
          Whether to mount a tmpfs on {file}`/tmp` during boot.
```
Comment (Scrumplex) [inline nixos/modules/system/boot/tmp.nix:28]: Maybe we can use this opportunity to also add inform about side effects, such as large builds (like the kernel) failing if there isn't enough space?

Last time I tried, a 16GB tmpfs wasn't enough space to build a kernel using Nix.

**Review by Scrumplex — APPROVED** [review]
Apart from my optional suggestion above, LGTM

File: nixos/modules/system/boot/tmp.nix [hunk 2]
```diff
@@ -3,62 +3,60 @@
...
+    boot.tmp.cleanOnBoot = mkOption {
       type = types.bool;
       default = false;
       description = lib.mdDoc ''
         Whether to delete all files in {file}`/tmp` during boot.
       '';
     };
 
-    boot.tmpOnTmpfs = mkOption {
+    boot.tmp.onTmpfs = mkOption {
```
Comment (Scrumplex) [inline nixos/modules/system/boot/tmp.nix:24]: Suggestion: `boot.tmp.mountTmpfs`

Comment (SuperSandro2000) [reply nixos/modules/system/boot/tmp.nix]: I forget to push that yesterday but I already renamed it to useTmpfs.

File: nixos/modules/system/boot/tmp.nix (hunk 1 above)
Comment (SuperSandro2000) [reply nixos/modules/system/boot/tmp.nix]: If you have a suggestion feel free to provide an improved text.

File: nixos/modules/system/boot/tmp.nix [hunk 3]
```diff
@@ -3,62 +3,62 @@
...
-      default = false;
-      description = lib.mdDoc ''
-         Whether to mount a tmpfs on {file}`/tmp` during boot.
-      '';
+      useTmpfs = mkOption {
+        type = types.bool;
+        default = false;
+        description = lib.mdDoc ''
+           Whether to mount a tmpfs on {file}`/tmp` during boot.
+        '';
```
Comment (Scrumplex) [inline nixos/modules/system/boot/tmp.nix:39]:
```suggestion
        description = lib.mdDoc ''
           Whether to mount a tmpfs on {file}`/tmp` during boot.
           Note that larger Nix builds can fail if the mounted tmpfs is not large enough.
        '';
```

Comment (SuperSandro2000) [reply nixos/modules/system/boot/tmp.nix]: Thanks! I based my change on this.

**Review by Scrumplex — APPROVED** [review]

**Review by RaitoBezarius — APPROVED** [review]
LGTM

**Pull request merged**